### DISCUSSION TITLE
editable kickstart UI

### DISCUSSION THREAD

**Discussion by cgwalters** [rhinstaller/anaconda#6013] [Ideas]
I know I wrote this somewhere in the past, but basically: chatting with someone who wanted a UI for `ostreecontainer`; while that would make sense, what I think would be *more* generally useful would be some button I could click in the GUI and get an editor for which I could input kickstart.

Especially if we made it easy to type in a URL to download kickstarts.

That would be a powerful generic "out" allowing ideally even mixing interactive configuration (partitioning, perhaps users) with fetching other automated config. One special case of that would let people do interactive partitioning but fetching a remote container image (and other kickstart config).

**Comment by vrothberg** [comment]
I love the proposal.  I'm in the process of running fedora-bootc on my workstation and would love an easier way to install my custom-built image.  Having the option to edit the kickstart in the UI would significantly improve the UX as I could point the installer directly to my OCI image on Quay.

**Comment by jkonecny12** [comment]
We were already thinking about something similar where we would utilize web UI of Anaconda. How it would work is that the installation with the Anaconda web UI will boot and the first screen will ask you if you want to upload your kickstart file to the installation.
This is especially useful in remote installations that you are not forced to have USB drive or external server to get your kickstart in. Also there is a possibility that we would be able to verify the correctness of the kickstart without a fatal failure. Currently, when you specify non-existing disk in kickstart, Anaconda is not able to do anything else except fail and ask for correction interactively or fail completely. However, with this approach there might be a possibility that you would fix your kickstart locally and just upload a new version. 
However, all of the above needs to be verified, I'm not sure how hard it could be to implement it that way.
